### PR TITLE
Fix \OC_App::getCurrentApp() when being called from CLI or phpunit

### PR DESCRIPTION
**Pull request by nickvergessen**
Fixes the failing tests in activity: https://github.com/nextcloud/activity/pull/777

### CODE REVIEW

File: lib/private/legacy/OC_App.php [hunk 1]
```diff
@@ -629,11 +629,20 @@ public static function getSettingsNavigation(): array {
 	 * @return string
 	 */
 	public static function getCurrentApp(): string {
+		if (\OC::$CLI) {
+			return '';
+		}
```
Comment (come-nc) [inline lib/private/legacy/OC_App.php:634]: Is this related? What does it fix?

File: lib/private/legacy/OC_App.php [hunk 2]
```diff
@@ -629,11 +629,20 @@ public static function getSettingsNavigation(): array {
...
+
 		$request = \OC::$server->getRequest();
 		$script = substr($request->getScriptName(), strlen(OC::$WEBROOT) + 1);
 		$topFolder = substr($script, 0, strpos($script, '/') ?: 0);
 		if (empty($topFolder)) {
-			$path_info = $request->getPathInfo();
+			try {
+				$path_info = $request->getPathInfo();
+			} catch (Exception $e) {
+				// Can happen from unit tests because the script name is `./vendor/bin/phpunit` or something a like then.
```
Comment (come-nc) [inline lib/private/legacy/OC_App.php:643]: Log the exception or catch something more specific.

Comment (nickvergessen) [reply lib/private/legacy/OC_App.php:643]: https://github.com/nextcloud/server/blob/07a9f34385a80570c2121f73d04bba12cfd39b3d/lib/private/AppFramework/Http/Request.php#L760
It's as specific as possible. But sure let me log it.

File: lib/private/legacy/OC_App.php (hunk 1 above)
Comment (nickvergessen) [reply lib/private/legacy/OC_App.php:634]: It fixes the case when this is called via cron/cli/occ
We don't need to do anything in that case, but it's not set to true when running tests

**Review by skjnldsv — APPROVED** [review]

**Review by come-nc — APPROVED** [review]

**Review by blizzz — APPROVED** [review]
